modules/statistics-export-csv/src/main/java/org/opencastproject/statistics/export/impl/StatisticsExportCSVImpl.java

```diff
@@ -0,0 +1,241 @@
+  private static final String[] header = {"ID", "Name", "Date", "Value"};
+  private static final String CFG_KEY_SERIES_TO_EVENT_PROVIDER_MAPPINGS = "series.to.event.provider.mappings";
+
+
+  private Map<String, String> seriesToEventProviderMapping = new HashMap<>();
+
+  private IndexService indexService;
+  private SecurityService securityService;
+  private StatisticsService statisticsService;
+
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void updated(Dictionary<String, ?> dictionary) {
+    final String providerMappings = (String) dictionary.get(CFG_KEY_SERIES_TO_EVENT_PROVIDER_MAPPINGS);
+    if (providerMappings == null) {
+      return;
+    }
+    try {
+      seriesToEventProviderMapping = (Map<String, String>) new JSONParser().parse(providerMappings);
+    } catch (ParseException e) {
+      throw new ConfigurationException(CFG_KEY_SERIES_TO_EVENT_PROVIDER_MAPPINGS, e);
+    }
+  }
+
+  public void activate(ComponentContext cc) {
+    logger.info("Activating Statistics Service");
+  }
+
+  public void deactivate(ComponentContext cc) {
+    logger.info("Deactivating Statistics Service");
+  }
+
+  public void setIndexService(IndexService indexService) {
+    this.indexService = indexService;
+  }
+
+  public void setStatisticsService(StatisticsService statisticsService) {
+    this.statisticsService = statisticsService;
+  }
+
+  public void setSecurityService(SecurityService securityService) {
+    this.securityService = securityService;
+  }
+
+  private static String formatDate(final String dateStr, DataResolution dataResolution, ZoneId zoneId) {
+    final LocalDateTime ldt = LocalDateTime.ofInstant(Instant.parse(dateStr), zoneId);
+    DateTimeFormatter formatter = null;
+    switch (dataResolution) {
+      case DAILY:
+        formatter = DateTimeFormatter.ofPattern("uuuu-MM-dd");
+        return formatter.format(ldt);
+      case WEEKLY:
+        formatter = DateTimeFormatter.ofPattern("uuuu-ww");
+        return formatter.format(ldt);
+      case MONTHLY:
+        formatter = DateTimeFormatter.ofPattern("uuuu-MM");
+        return formatter.format(ldt);
+      case YEARLY:
+        formatter = DateTimeFormatter.ofPattern("uuuu");
+        return formatter.format(ldt);
+      default:
+        throw new IllegalStateException("Unexpected value: " + dataResolution);
+    }
+  }
+
+
+  @Override
+  public String getCSV(StatisticsProvider provider, String resourceId, Instant from, Instant to, DataResolution
+      dataResolution, AbstractSearchIndex index, ZoneId zoneId) throws SearchIndexException, UnauthorizedException,
+      NotFoundException {
+    if (!(provider instanceof TimeSeriesProvider)) {
+      throw new IllegalStateException("CSV export not supported for provider of type " + provider.getClass().getName());
+    }
+    final StringWriter stringWriter = new StringWriter();
+    try (CSVPrinter printer = CSVFormat.EXCEL.withHeader(header).print(stringWriter)) {
```

Comment:
I haven't actually run this code yet, but this looks like we're spitting out the binary CSV format for Excel, correct?  Is that a hard requirement?  I, personally, hate getting not-actually-comma-separated data as a CSV, so I would be tempted to make this spit out true CSV data instead.

Reply:
Here is a description of the different formats supported by apache commons csv: 

http://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html

I changed it to RFC4180 to avoid confusion but this doesn't make a difference in our case. The output is a plain text CSV file which can be used with excel or any other tool. However, since there is no real standard for CSV, Some tools may require the file to be edited before it can be imported.